src/main/java/org/spongepowered/common/data/processor/data/item/ItemDyeColorDataProcessor.java

```diff
@@ -22,33 +22,50 @@
  * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
  * THE SOFTWARE.
  */
-package org.spongepowered.common.data.processor.value.item;
+package org.spongepowered.common.data.processor.data.item;
 
+import net.minecraft.block.Block;
+import net.minecraft.init.Blocks;
 import net.minecraft.init.Items;
 import net.minecraft.item.EnumDyeColor;
+import net.minecraft.item.Item;
 import net.minecraft.item.ItemStack;
 import org.spongepowered.api.data.DataTransactionResult;
 import org.spongepowered.api.data.key.Keys;
+import org.spongepowered.api.data.manipulator.immutable.ImmutableDyeableData;
+import org.spongepowered.api.data.manipulator.mutable.DyeableData;
 import org.spongepowered.api.data.type.DyeColor;
 import org.spongepowered.api.data.type.DyeColors;
 import org.spongepowered.api.data.value.ValueContainer;
 import org.spongepowered.api.data.value.immutable.ImmutableValue;
 import org.spongepowered.api.data.value.mutable.Value;
-import org.spongepowered.common.data.processor.common.AbstractSpongeValueProcessor;
-import org.spongepowered.common.data.processor.data.DyeableDataProcessor;
+import org.spongepowered.common.data.manipulator.mutable.SpongeDyeableData;
+import org.spongepowered.common.data.processor.common.AbstractItemSingleDataProcessor;
 import org.spongepowered.common.data.value.SpongeValueFactory;
 
 import java.util.Optional;
 
-public class ItemDyeColorValueProcessor extends AbstractSpongeValueProcessor<ItemStack, DyeColor, Value<DyeColor>> {
+public class ItemDyeColorDataProcessor extends AbstractItemSingleDataProcessor<DyeColor, Value<DyeColor>, DyeableData, ImmutableDyeableData> {
 
-    public ItemDyeColorValueProcessor() {
-        super(ItemStack.class, Keys.DYE_COLOR);
+    public ItemDyeColorDataProcessor() {
+        super(x -> isDyeable(x.getItem()), Keys.DYE_COLOR);
     }
 
-    @Override
-    protected boolean supports(ItemStack container) {
-        return DyeableDataProcessor.isDyeable(container.getItem());
+    public static boolean isDyeable(Item item) {
+        if (item.equals(Items.DYE)) {
+            return true;
+        }
+
+        final Block block = Block.getBlockFromItem(item);
+
+        return block.equals(Blocks.WOOL)
+                || block.equals(Blocks.STAINED_GLASS)
+                || block.equals(Blocks.CARPET)
+                || block.equals(Blocks.STAINED_GLASS_PANE)
+                || block.equals(Blocks.STAINED_HARDENED_CLAY)
+                || block.equals(Blocks.CONCRETE)
+                || block.equals(Blocks.CONCRETE_POWDER)
+                || block.equals(Blocks.BED);
```

Comment:
Are bed blocks and bed items paired? I suggest using `Items.BED` in checks.

Reply:
You are correct. Let me fix that.